helpers/analytics_helpers.rb

```diff
@@ -0,0 +1,42 @@
+module AnalyticsHelpers
+  def urlize(string)
+    string.downcase.gsub(" ", "_").gsub("-", "_")
+  end
+
+  def build_event(data, attributes = [], output = {})
+    data.sort_by { |k, _v| k["name"] }.each do |item|
+      if item["value"]
+        case item["value"]
+        when String
+          output[item["name"]] = item["value"]
+        when Array
+          output[item["name"]] = build_event(item["value"], attributes)
+        end
+      else
+        attribute = attributes.select { |x| x["name"] == item["name"] }
```

Comment:
Because you always call `.first` on the result, you could replace this `.select` with `.find` and save yourself the calls to `.first`